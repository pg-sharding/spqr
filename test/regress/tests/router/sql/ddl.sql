\c spqr-console

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid2 FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;

ALTER DISTRIBUTION ds1 ATTACH RELATION table_1 DISTRIBUTION KEY id;
ALTER DISTRIBUTION ds1 ATTACH RELATION table_2 DISTRIBUTION KEY id;

\c regress

CREATE TABLE table_1(id INT PRIMARY KEY);
CREATE TABLE table_2(id INT PRIMARY KEY);

BEGIN;
ALTER TABLE "table_1" RENAME TO "tmp";
ALTER TABLE "table_2" RENAME TO "table_1";
ALTER TABLE "tmp" RENAME TO "table_2";
COMMIT;

DROP TABLE table_1 CASCADE;
DROP TABLE table_2;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;