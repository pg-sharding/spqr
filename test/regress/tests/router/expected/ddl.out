\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE krid1 FROM 1 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 1
(1 row)

CREATE KEY RANGE krid2 FROM 101 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 101
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION table_1 DISTRIBUTION KEY id;
        attach table        
----------------------------
 relation name   -> table_1
 distribution id -> ds1
(2 rows)

ALTER DISTRIBUTION ds1 ATTACH RELATION table_2 DISTRIBUTION KEY id;
        attach table        
----------------------------
 relation name   -> table_2
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE table_1(id INT PRIMARY KEY);
NOTICE: send query to shard(s) : sh1,sh2
CREATE TABLE table_2(id INT PRIMARY KEY);
NOTICE: send query to shard(s) : sh1,sh2
BEGIN;
ALTER TABLE "table_1" RENAME TO "tmp";
NOTICE: send query to shard(s) : sh1,sh2
ALTER TABLE "table_2" RENAME TO "table_1";
ALTER TABLE "tmp" RENAME TO "table_2";
COMMIT;
DROP TABLE table_1 CASCADE;
NOTICE: send query to shard(s) : sh1,sh2
DROP TABLE table_2;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

